exo/inference/test_dummy_inference_engine.py

```diff
@@ -0,0 +1,30 @@
+import asyncio
+import pytest
+from exo.inference.DummyInferenceEngine import DummyInferenceEngine  # Adjust the path to where you implemented DummyInferenceEngine
+
+@pytest.mark.asyncio
+async def test_dummy_inference_engine_static():
+    # Test with static output
+    dummy_engine = DummyInferenceEngine(output_type="static", output_value=[1, 2, 3], latency_mean=0.2, latency_stddev=0.05)
+    await dummy_engine.run_inference()  # Simulate inference, check for errors
+    assert dummy_engine.output_value == [1, 2, 3], "The static output should match the provided value."
+
+
+@pytest.mark.asyncio
+async def test_dummy_inference_engine_random():
+    # Test with random output
+    dummy_engine = DummyInferenceEngine(output_type="random", output_shape=(2, 2), latency_mean=0.1, latency_stddev=0.01)
+    await dummy_engine.run_inference()  # Simulate inference, check for errors
+    output = dummy_engine.output_value
+    assert isinstance(output, list), "Output should be a list."
+    assert len(output) == 2, "Output should have the specified shape."
+
+
+@pytest.mark.asyncio
+async def test_dummy_inference_engine_latency():
+    # Test that latency is within expected range
+    dummy_engine = DummyInferenceEngine(output_type="static", output_value=[1], latency_mean=0.1, latency_stddev=0.02)
+    start_time = asyncio.get_event_loop().time()
+    await dummy_engine.run_inference()
+    elapsed_time = asyncio.get_event_loop().time() - start_time
+    assert 0.06 <= elapsed_time <= 0.14, f"Expected latency to be around 0.1s, but got {elapsed_time}s."
```

Comment:
This is going to fail sometimes. Either make this so that probability of this failing is 1 in 1e18 (with multiple runs or just tuning stddev) or make stddev zero.

Reply:
yep its quite sensitive. thought of let it run for a few 100k cycles to get a better idea of mean and stddev and also on different machines, but i guess your suggestion is the more pragmatic approach.